statsmodels/tsa/automatic/sarimax.py

```diff
@@ -3,24 +3,92 @@
 import pandas as pd
 import statsmodels.api as sm
 
-def auto_order(endog, criteria='aic', d=0, max_order=(3, 3), **spec=None):
+
+def auto_order(
+                endog, criteria='aic', d=0, max_order=(3, 3),
+                stepwise=False, **spec):
     """Auto order function for SARIMAX models."""
-    initp, initq = max_order[0], max_order[1]
-    aic_matrix = np.zeros((initp, initq))
-    aic_matrix[0, 0] = np.inf
-    for p in range(max_order[0]):
-        for q in range(max_order[1]):
-            if p == 0 and q == 0:
-                continue
-            # fit  the model
-            mod = sm.tsa.statespace.SARIMAX(endog, order=(p, d, q), **spec)
+    if not stepwise:
+        aic_matrix = np.zeros((max_order[0], max_order[1]))
+        for p in range(max_order[0]):
+            for q in range(max_order[1]):
+                # fit  the model
+                mod = sm.tsa.statespace.SARIMAX(endog, order=(p, d, q), **spec)
+                res = mod.fit(disp=False)
+                aic_matrix[p, q] = res.aic
+                # print(res.aic)
+        # print(aic_matrix)
+        min_aic = aic_matrix.min()
+        # print(min_aic)
+        p, q = np.where(aic_matrix == min_aic)
+        return p, q
+    else:
+        """stepwise algorithm for auto order"""
+        aic_vals = np.zeros(4)
+        order_init = {
+                        0: [2, 2],
+                        1: [0, 0],
+                        2: [1, 0],
+                        3: [0, 1]
+                    }
+        for model in range(4):
+            mod = sm.tsa.statespace.SARIMAX(
+                endog, order=(order_init[model][0], d, order_init[model][1]),
+                **spec)
+            res = mod.fit(disp=False)
+            aic_vals[model] = res.aic
+        print(aic_vals)
+        min_aic = aic_vals.min()
+        model = int(np.where(aic_vals == min_aic)[0])
+        p, q = order_init[model][0], order_init[model][1]
+        # only p varies by +1
+        if(p+1 <= max_order[0]):
+            mod = sm.tsa.statespace.SARIMAX(endog, order=(p+1, d, q), **spec)
+            res = mod.fit(disp=False)
+            # print(res.aic)
+            if (res.aic < min_aic):
+                p += 1
+                min_aic = res.aic
+        # only p varies by -1
+        if(p-1 >= 0):
```

Comment:
If the previous case `(p+1)` improved the aic, then this will test the original `p` value again.

E.g. if the `order_init` was minimized with `p=1, q=0` and then `p+1=2` was better than `p = 1`, this section will test `p = 1` again.

Reply:
Yes, I understand that. It should be checking for p=0. I'll add a new variable that'll keep track of the changes in p and finally assign it in the end.